lib/tbd/geo.rb

```diff
@@ -357,22 +362,82 @@ def properties(surface = nil, argh = {})
     surface.subSurfaces.sort_by { |s| s.nameString }.each do |s|
       next if poly(s).empty?
 
-      id       = s.nameString
-      valid    = s.vertices.size == 3 || s.vertices.size == 4
-      log(ERR, "Skipping '#{id}': vertex # 3 or 4 (#{mth})") unless valid
-      next                                                   unless valid
-
-      vec      = s.vertices
-      area     = s.grossArea
-      mult     = s.multiplier
-      typ      = s.subSurfaceType.downcase
-      type     = :skylight
-      type     = :window       if typ.include?("window" )
-      type     = :door         if typ.include?("door"   )
-      glazed   = type == :door && typ.include?("glass"  )
-      tubular  =                  typ.include?("tubular")
-      domed    =                  typ.include?("dome"   )
-      unhinged = false
+      id  = s.nameString
+      typ = surface.surfaceType.downcase
+
+      unless (3..4).cover?(s.vertices.size)
+        log(ERR, "Skipping '#{id}': vertex # 3 or 4 (#{mth})")
+        next
+      end
+
+      vec  = s.vertices
+      area = s.grossArea
+      mult = s.multiplier
+
+      # An OpenStudio subsurface has a "type" (string), either defaulted during
+      # initialization or explicitely set by the user (from a built-in list):
+      #
+      #   OpenStudio::Model::SubSurface.validSubSurfaceTypeValues
+      #   - "FixedWindow"
+      #   - "OperableWindow"
+      #   - "Door"
+      #   - "GlassDoor"
+      #   - "OverheadDoor"
+      #   - "Skylight"
+      #   - "TubularDaylightDome"
+      #   - "TubularDaylightDiffuser"
+      typ = s.subSurfaceType.downcase
+
+      # An OpenStudio default subsurface construction set can hold unique
+      # constructions assigned for each of these admissible types. In addition,
+      # type assignment determines whether frame/divider attributes can be
+      # linked to a subsurface (this shortlist has evolved between OpenStudio
+      # releases). Type assignment is relied upon when calculating (admissible)
+      # fenestration areas. TBD also relies on OpenStudio subsurface type
+      # assignment, with resulting TBD tags being a bit more concise, e.g.:
+      #
+      #   - :window includes "FixedWindow" and "OperableWindow"
+      #   - :door includes "Door", "OverheadWindow" and "GlassDoor"
+      #     ... a (roof) access roof hatch should be assigned as a "Door"
+      #   - :skylight includes "Skylight", "TubularDaylightDome", etc.
```

Comment:
Documentation/justification in light of ASHRAE 2022 requirements on linear thermal bridging.